Alkiiro/refactor-tractor
# Initial inheritance Repo parent class

What's this PR do?
This PR update created the initial file for our new Repo Parent Class.  I added the data ,id, and user properties which were included in all of the Sleep,Activity, and Hydration classes.  I also added a `getFitData` method which will end up removing the repetitive method included in all 3 child classes.  I have only updated this for the Sleep child class currently.  This PR creates the Repo parent class, testing for the first method, and updates to the Sleep class for all Sleep properties and new variable names changed in Sleep class methods.  I have also removed the Sleep getUserData method since it is now being called in parent class.

Where should the reviewer start?
I would take a look at the Repo properties and Sleep properties to see the repetition and how parent class is able to assist child with providing those properties.  Also the `getFitData` method will now take care of the repetitive `getUserData` methods included in Sleep,Activity,and Hydration.

How should this be manually tested?
I would pull the repo down and console.log to make sure data is being pulled correctly.  All tests passed on my end.

Any background context you want to provide?
This Repo class will continue to be built out to provide more methods for the repetitive methods in child classes.  
@brittanystoroz 
@khalidwilliams
